File: typescript/functions/array_operations.ts
const numbers: number[] = [1, 2, 3, 4, 5];
const fruits: string[] = ['apple', 'banana', 'cherry'];

console.log(`Numbers: ${numbers}`);
console.log(`Length: ${numbers.length}`);

const sum: number = numbers.reduce((a, b) => a + b, 0);
console.log(`Sum: ${sum}`);

const max: number = Math.max(...numbers);
console.log(`Max: ${max}`);

const min: number = Math.min(...numbers);
console.log(`Min: ${min}`);

numbers.push(6);
console.log(`After push: ${numbers}`);

numbers.pop();
console.log(`After pop: ${numbers}`);

const squared: number[] = numbers.map((x) => x ** 2);
console.log(`Squared: ${squared}`);

const filtered: number[] = numbers.filter((x) => x % 2 === 0);
console.log(`Even numbers: ${filtered}`);

const combined: (number | string)[] = [...numbers, ...fruits];
console.log(`Combined: ${combined}`);

File: java/functions/ArrayOperations.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

public class ArrayOperations {
    public static void main(String[] args) {
        ArrayList<Integer> numbers = new ArrayList<>(Arrays.asList(1, 2, 3, 4, 5));

        System.out.println("Numbers: " + numbers);
        System.out.println("Size: " + numbers.size());

        int sum = numbers.stream().mapToInt(Integer::intValue).sum();
        System.out.println("Sum: " + sum);

        int max = Collections.max(numbers);
        System.out.println("Max: " + max);

        int min = Collections.min(numbers);
        System.out.println("Min: " + min);

        numbers.add(6);
        System.out.println("After add: " + numbers);

        numbers.remove(numbers.size() - 1);
        System.out.println("After remove: " + numbers);

        ArrayList<Integer> squared = new ArrayList<>();
        for (int num : numbers) {
            squared.add(num * num);
        }
        System.out.println("Squared: " + squared);
    }
}
